thrift_rpc/recorder.go

```diff
@@ -385,6 +387,19 @@ func (r *Recorder) Flush() {
 	}
 }
 
+func (r *Recorder) Close() error {
+	r.lock.Lock()
+	closech := r.closech
+	r.closech = nil
+	r.lock.Unlock()
+
+	if closech != nil {
+		close(closech)
+	}
+
+	return nil
```

Comment:
To clarify - I realize the original comment was in the thrift code but I believe it applies to the gRPC path as well. I don't see any place where we are waiting for the final flush to complete. We do a final flush asynchornously (in reportLoop) but we are not waiting for it in `CloseTracer()`. In fact, we are closing the `ClientConn` immediately so the last `Flush` is likely to just exit. Am I missing something?